tests/Type/ScalarSerializationTest.php

```diff
@@ -204,11 +204,11 @@ public function testSerializesOutputAsID(): void
     public function badIDValues(): iterable
     {
         return [
-            [new stdClass(), 'ID cannot represent value: instance of stdClass'],
-            [true, 'ID cannot represent value: true'],
-            [false, 'ID cannot represent value: false'],
-            [-1.1, 'ID cannot represent value: -1.1'],
-            [['abc'], 'ID cannot represent value: ["abc"]'],
+            [new stdClass(), 'ID cannot represent a non-string and non-integer value: instance of stdClass'],
```

Comment:
```suggestion
            [new stdClass(), 'ID cannot represent a non-string or non-integer value: instance of stdClass'],
```
?

Reply:
all of them I guess

Reply:
This comes from the reference implementation, and I think it is fine: the value is both a non-string *and* a non-integer. Or would be appropriate if the statement were to list allowed values, e.g. `ID can only represent string or integer, got: `.